### ISSUE TITLE
Pre-release cleanup

### ISSUE THREAD

**Issue by thedataking** [memorysafety/rav1d#459]
The repository contains artifacts that are helpful for migration but are not necessary to build or maintain the final deliverable. This includes:

- The dav1d C headers and sources
- Scripts added during migration (e.g. `retranspile.sh`)
- Changes to `.gitignore` to facilitate migration
- The dav1d build systems

After some discussion, the plan of record is to store these artifacts on a side branch and delete them from the main branch. Removing the meson build files is nice to have since it is contingent on tests being run by cargo, not meson.

We also need to make sure the README is updated to accurately reflect maturity, build steps,  prerequisites, etc.
